src/main/java/codechicken/nei/util/NBTJson.java

```diff
@@ -163,18 +188,43 @@ public static NBTBase toNbt(JsonElement jsonElement) {
         } else if (jsonElement instanceof JsonObject) {
             // NBTTagCompound
             final JsonObject jsonObject = (JsonObject) jsonElement;
-            NBTTagCompound nbtTagCompound = new NBTTagCompound();
-            for (Map.Entry<String, JsonElement> jsonEntry : jsonObject.entrySet()) {
-                nbtTagCompound.setTag(jsonEntry.getKey(), toNbt(jsonEntry.getValue()));
+            final NBTBase custom = restoreEmptyList(jsonObject);
+
+            if (custom == null) {
+                NBTTagCompound nbtTagCompound = new NBTTagCompound();
+
+                for (Map.Entry<String, JsonElement> jsonEntry : jsonObject.entrySet()) {
+                    nbtTagCompound.setTag(jsonEntry.getKey(), toNbt(jsonEntry.getValue()));
+                }
+    
+                return nbtTagCompound;
+            } else {
+                return custom;
             }
-            return nbtTagCompound;
+
         }
         
         throw new IllegalArgumentException("Unhandled element " + jsonElement);
     }
-    public static byte[] toByteArray(IntStream stream) {
-        return stream.collect(ByteArrayOutputStream::new, (baos, i) -> baos.write((byte) i), (baos1, baos2) -> baos1.write(baos2.toByteArray(), 0, baos2.size()))
-            .toByteArray();
+
+    protected static JsonObject createEmptyList(Object obj)
```

Comment:
The script could not correctly restore empty NBTTagList, NBTTagIntArray & NBTTagByteArray. this hack is designed to solve this problem